terraform-dev/redis.tf

```diff
@@ -0,0 +1,17 @@
+resource "google_project_iam_binding" "redis_service_agent_binding" {
+  project       = var.project_id
+  role          = "roles/redis.serviceAgent"
+  members       = ["serviceAccount:service-${var.project_number}@cloud-redis.iam.gserviceaccount.com"]
+}
```

Comment:
Project-level IAM is configured in `main-gcp.tf` in the `google_project_iam_policy.project` resource.  I think this will conflict with that, but even if it doesn't, it's preferable to use the policy.  We use policies rather than bindings, because policies mean that terraform removes any IAM configurations that it doesn't know about, so we can use terraform to clean up after ourselves if we have given someone a role manually via the web console and then forgotten to revoke it.